aws_jupyter_proxy/awsproxy.py

```diff
@@ -48,10 +47,9 @@ def get_service_info(
         region_name=region,
     )
 
-    if (
-        endpoint_override
-        and validators.url(endpoint_override)
-        and re.compile(".*\.(aws.dev|amazonaws.com)$").match(endpoint_override)
+    if endpoint_override and re.fullmatch(
+        r"https:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.(aws.dev|amazonaws.com)\b",
```

Comment:
Noting here as well: Can we add a negative test case where this regex rejects the input endpoint_override?

Reply:
I am adding two negative tests to next revision.. one where were are using improper http, another where we are pointing to a non-allowlisted destination.